### PR TITLE
Add salt and rabbitmq test suites

### PR DESCRIPTION
**Pull request by okurz**
* Add basic salt test suite
* Add rabbitmq test suite

screenshot of local verification on Tumbleweed:
openqa_rabbitmq_salt

### CODE REVIEW

**Comment by sysrich** [comment]
Adding new tests and fixing snapshots? This seems like two very different topics, shouldn't that be two different pull requests?

SCNR ;)

File: tests/console/salt.pm [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+use base "consoletest";
+use strict;
+use testapi;
+
+sub run() {
+    select_console 'user-console';
+    assert_script_sudo("zypper -n in salt-master salt-minion");
+    assert_script_sudo("systemctl start salt-master");
+    assert_script_run("systemctl status salt-server");
```
Comment (sysrich) [inline tests/console/salt.pm]: I think you need to do s/salt-server/salt-master

File: tests/console/rabbitmq.pm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+use base "consoletest";
+use strict;
+use testapi;
+
+sub run() {
+    select_console 'user-console';
+    assert_script_sudo("zypper -n in rabbitmq-server");
+    assert_script_sudo("systemctl start rabbitmq-server");
+    assert_script_run("systemctl status rabbitmq-server");
```
Comment (sysrich) [inline tests/console/rabbitmq.pm]: assert_script_run on a systemctl status will not show the journal logs of that service. Is that intentional?

If not, please use assert_script_sudo

**Comment by okurz** [comment]
split out the "fix consoletests part". Thanks for the suggestion. I was just trying to tackle these issues :-)

**Comment by okurz** [comment]
rabbitmq fails on Tumbleweed because "ssl" is missing for python2 but python-pika is only available for python2. I would say this is a missing package problem in Tumbleweed. How should we proceed? It works on Leap42.2, though.

**Comment by sysrich** [comment]
File bugs, merge the test but run it as an extratest as not to block TW
releases?

**Comment by okurz** [comment]
just merged as discussed with sysrich. Will handle the missing package on TW based on the fail. Also, for Leap42.2 we are missing extratests.

@DimStar77 Tumbleweed seems to miss `python2-pyOpenSSL`

@lnussel Leap42.2 are missing all the "extra tests": Compare to https://openqa.opensuse.org/tests/overview?distri=opensuse&groupid=1&version=Tumbleweed and look for "extra_tests_on_…". Can we add them?

**Comment by DimStar77** [comment]
On Thu, 2016-07-21 at 08:32 -0700, Oliver Kurz wrote:
> > > just merged as discussed with sysrich. Will handle the missing
package on TW based on the fail. Also, for Leap42.2 we are missing
extratests.
> 
> @DimStar77 Tumbleweed seems to miss python2-pyOpenSSL

python-pyOpenSSL has been part of the distribution since 2011,
currently in version 16.0.0
> > 
> > > 
> > >

**Pull request merged**